intera_interface/src/motion_interface/utility_functions.py

```diff
@@ -0,0 +1,91 @@
+#! /usr/bin/env python
+
+# Copyright (c) 2016-2017, Rethink Robotics Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import os
+import math
+import errno
+from intera_core_msgs.srv import (
+    SolvePositionFK,
+    SolvePositionFKRequest,
+    SolvePositionIK,
+    SolvePositionIKRequest,
+)
+import rospy
+from geometry_msgs.msg import (
+    PoseStamped,
+    Pose,
+    Point,
+    Quaternion,
+)
+import csv
+from std_msgs.msg import Header
+from sensor_msgs.msg import JointState
+from copy import deepcopy
+
+def get_formatted_decimal_string(i, n):
+    """
+    @param i: index
+    @param n: total number of indicies
+    @return: str(i) with correct number of padding zeros
+    """
+    n_digits = len(str(n))
+    format_string = '{:0>' + str(n_digits) + 'd}'
```

Comment:
You likely don't want the "d" in the string formatting. It restricts you to only integers, which won't have decimal digitas anyway. Rewritten for forward compatibility with python 3.x:
```
format_string = '{{:0>{}}}'.format(n_digits)
```